Allow FuseMountName for chansrv to be absolute path

matt335672 opened the pull request:
This PR primarily addresses #1048 (Drive redirection location).

It also contains:-
- a fix for the first of the two issues mentioned in #1658 (xrdp-chansrv become defunct in docker while file copy) 
- a way to disable the FUSE filesystem at runtime if the system was built with `--enable-fuse`.

Documentation for the new features has been added to `sesman.ini` and the manpage `xrdp-sesman.ini(5)`.

The code has been largely tested by @hmakmur, but, as ever, could do with a second pair of eyes.

An extra module `string_calls.c` has been added to contain the `g_format_info_string()` function. Over time, some other string functions which are not operating-system specific could be moved here from `os_calls.c`.

metalefty commented:
I haven't tested yet but overall LGTM.

metalefty commented on common/os_calls.c:
```diff
@@ -3307,6 +3307,17 @@ g_setsid(void)
 
+/*****************************************************************************/
+int
+g_getusername(char *name, unsigned int len)
```
I'm fine with this function name. BTW, why is this named `getusername` rather than `getlogin`?

matt335672 replied:
Good point - I think it was where I was coming from when I was putting the calling function together. I agree that `g_getlogin` is more expected here, so I've changed it.

metalefty commented on docs/man/sesman.ini.5.in:
```diff
@@ -258,6 +277,21 @@ files on your redirected drives. This may not be approprate for all
+.P
+-  drive redirection
+.P
+-  cutting-and-pasting of files.
```
Trivial point. Remove periods at the end. Otherwise put periods to both?

matt335672 replied:
Trivial's fine by me. Period removed.

metalefty commented on sesman/chansrv/chansrv_config.c:
```diff
@@ -266,18 +272,28 @@ config_read(int use_logger, const char *sesman_ini)
 
+/******************************************************************************/
+static const char *
+bool2text(int value)
```
I'm fine with this at the moment but this function can be used in general contexts not only `chansrv`.
I think somewhere in `common/` will be a more appropriate place.  It's not a `os_calls` though.

matt335672 replied:
Absolutely, and there's one in log.c. I've moved it to the new string_calls.c file. We could maybe move some other stuff out of os_calls.c into here too. What do you think?

metalefty replied:
I agree.

matt335672 commented:
Hadn't noticed this one was no longer mergeable - it now is.

I've squashed the original commits into 1, and added another one which moves g_text2bool() into string_calls, where it joins g_bool2text(). So that's now consistent.

@metalefty - there's a bunch of string calls in os_calls.h which could be moved in to string_calls.h:-

https://github.com/neutrinolabs/xrdp/blob/bfe76e0499fd2c02eb2c97d49dce79ffa2262a19/common/os_calls.h#L122-L142

I could move them now. Quite a few files might need `#include "string_calls.h"` adding, but I think that's a pretty minimal change, provided it stays as a single commit. Do you agree?

metalefty commented:
@matt335672

Hi, sorry for the delay. Of course, a separate PR is a good idea but let's add a single commit to this PR to move them.
Merge it when you finished.

metalefty commented on docs/man/sesman.ini.5.in:
```diff
@@ -258,6 +277,21 @@ files on your redirected drives. This may not be approprate for all
+.P
+-  drive redirection
+.P
+-  cutting-and-pasting of files
```
"copy and paste" rather than "cut and paste".

matt335672 replied:
Good catch!

matt335672 replied:
Got some family commitments for a couple of days. I'll fix the above, do a squash and then add a separate commit for the string function move as soon as I can.

metalefty commented on common/string_calls.h:
```diff
@@ -0,0 +1,104 @@
+
+int      g_strlen(const char* text);
+const char *g_strchr(const char *text, int c);
+char*    g_strcpy(char* dest, const char* src);
```
Trivial point, not caused by your changes. I found `char *name` and `char* name` are mixed. Let's fix it someday in the future.

matt335672 replied:
I'll pick it up with astyle before I commit to devel
